api/src/main/java/javax/application/converter/BigDecimalConverter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javax.application.converter;
+
+import javax.application.formatter.BigDecimalFormatter;
+import javax.application.formatter.Formatter;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * @author Andres Almiray
+ */
+public class BigDecimalConverter extends AbstractFormattingConverter<BigDecimal> implements NumberConverter<BigDecimal> {
```

Comment:
I think it is wrong to already provide this implementations by the SPEC. Have a look at Java Bean validation spec. All the constraints annotations like @NotNull are part of the spec but the spec do not contain any implementation for checking this constraints. Even if the spec could easily provide implementations for that they do not add them. 

All this converter impl. should be provided by frameworks that implement the spec

Reply:
well, both UnitsOfMeasurements and JavaMoney provide an SPI package but no default impls. I suppose we could take out the default converter implementations (and the whole formatter package with it) and let implementors decide how they want to implement the spec.

Alternatively, this code could be part of a `jsr377-extras` binary. @keilw @atsticks what are the rules governing the bp released of JavaMoney, also the additional binaries provides by UnitsOfMeasurements? Perhaps it would be better to move this discussion to the mailing list

Reply:
Yes, I guess that should work. JSR 354 has a few more static factories like `Monetary`, JSR 363 for a smaller footprint only has one (`ServiceProvider`) but they both work via SPI through implementations found on the classpath. JSR 363 now has plenty of extension modules for different unit systems, so extensibility sounds like a good idea here, too.